src/Tasks/Al.cs

```diff
@@ -305,7 +305,12 @@ protected override string GenerateFullPathToTool()
 
             if (String.IsNullOrEmpty(pathToTool) || !FileSystems.Default.FileExists(pathToTool))
             {
-                pathToTool = SdkToolsPathUtility.GeneratePathToTool(SdkToolsPathUtility.FileInfoExists, ProcessorArchitecture.CurrentProcessArchitecture, SdkToolsPath, ToolExe, Log, true);
+                // The bitness of al.exe should match the platform being built
+                string archToLookFor =  Platform.Equals("x86", StringComparison.OrdinalIgnoreCase) ? Platform :
+                                        Platform.Equals("x64", StringComparison.OrdinalIgnoreCase) ? ProcessorArchitecture.AMD64 : // x64 maps to AMD64 in GeneratePathToTool
+                                        ProcessorArchitecture.CurrentProcessArchitecture;
+
+                pathToTool = SdkToolsPathUtility.GeneratePathToTool(SdkToolsPathUtility.FileInfoExists, archToLookFor, SdkToolsPath, ToolExe, Log, true);
```

Comment:
Is this one of those cases where SdkToolsPathUtility.FileInfoExists is secretly slower than `f => SdkToolsPathUtility.FileInfoExists(f)`?

Reply:
😮 can you post an example of us making that change?

Reply:
https://github.com/dotnet/msbuild/pull/6983

Reply:
Yeah, this thing is devious. It's not a big deal here since this isn't a hot loop but please go ahead and fix.